tests/integration/test_task.py

```diff
@@ -80,3 +83,229 @@ def _run(self, input_files):
+           "({}): file does not exist.".format(script_file, script_file))
+    assert ept == str(err_msg.value)
+
+
+def _get_diagnostic_tasks(tmp_path, diagnostic_text, extension):
+    """Assemble Python diagnostic tasks of DiagnosticTasks."""
+    diag = 'diag_cow.' + extension
+    diag_script = tmp_path / diag
+    diag_output_dir = tmp_path / 'mydiag'
+    diag_run_dir = diag_output_dir / 'run_dir'
+    diag_plot_dir = diag_output_dir / 'plot_dir'
+    diag_work_dir = diag_output_dir / 'work_dir'
+    diag_settings = {
+        'run_dir': diag_run_dir.as_posix(),
+        'plot_dir': diag_plot_dir.as_posix(),
+        'work_dir': diag_work_dir.as_posix(),
+        'profile_diagnostic': False,
+        'exit_on_ncl_warning': False
+    }
+
+    with open(diag_script, "w") as fil:
+        fil.write(diagnostic_text)
+
+    task = DiagnosticTask(
+        name='task0',
+        ancestors=None,
+        script=diag_script.as_posix(),
+        settings=diag_settings,
+        output_dir=diag_output_dir.as_posix(),
+    )
+
+    return {task}
+
+
+def test_py_diagnostic_run_sequential_task_fails(monkeypatch, tmp_path):
+    """Run DiagnosticTask sequentially with bad Python script."""
+    diagnostic_text = "import os\n\nprint(cow)"
+    tasks = _get_diagnostic_tasks(tmp_path, diagnostic_text, 'py')
+
+    def _run(self, input_filesi=[]):
+        print(f'running task {self.name}')
+
+    monkeypatch.setattr(BaseTask, '_run', _run)
+
+    with pytest.raises(DiagnosticError) as err_mssg:
+        _run_tasks_sequential(tasks)
+    exp_mssg = "diag_cow.py failed with return code 1"
+    assert exp_mssg in str(err_mssg.value)
+
+
+def test_py_diagnostic_run_parallel_task_fails(monkeypatch, tmp_path):
+    """Run DiagnosticTask parallel with bad Python script."""
+    diagnostic_text = "import os\n\nprint(cow)"
+    tasks = _get_diagnostic_tasks(tmp_path, diagnostic_text, 'py')
+
+    def _run(self, input_filesi=[]):
+        print(f'running task {self.name}')
+
+    monkeypatch.setattr(BaseTask, '_run', _run)
+
+    with pytest.raises(DiagnosticError) as err_mssg:
+        for task in tasks:
+            task.run()
+    exp_mssg = "diag_cow.py failed with return code 1"
+    assert exp_mssg in str(err_mssg.value)
+
+
+def test_py_diagnostic_run_parallel_task(monkeypatch, tmp_path):
+    """Run DiagnosticTask in parallel with OK Python script."""
+    diagnostic_text = "import os\n\nprint('cow')"
+    tasks = _get_diagnostic_tasks(tmp_path, diagnostic_text, 'py')
+
+    def _run(self, input_filesi=[]):
+        print(f'running task {self.name}')
+
+    monkeypatch.setattr(BaseTask, '_run', _run)
+
+    for task in tasks:
+        task.run()
+
+
+def test_ncl_diagnostic_run_parallel_task_fails(monkeypatch, tmp_path):
+    """Run DiagnosticTask in parallel with OK NCL script."""
+    diagnostic_text = "cows on the [river]"
+
+    def _run(self, input_filesi=[]):
+        print(f'running task {self.name}')
+
+    if shutil.which('ncl') is not None:
```

Comment:
It would be better to skip the test if ncl is not installed, rather than silently passing it. You can use [`pytest.mark.skipif`](https://docs.pytest.org/en/latest/skipping.html#id1) to do that.